components/custom_status/expiry_menu.tsx

```diff
@@ -0,0 +1,104 @@
+}
+
+const ExpiryMenu: React.FC<Props> = (props: Props) => {
+    const {expiry, handleExpiryChange} = props;
+    const expiryMenuItems: { [key: string]: ExpiryMenuItem } = {
+        'dont-clear': {
+            text: "Don't clear",
+            value: "Don't clear",
+            localizationId: 'dont_clear',
+        },
+        'thirty-minutes': {
+            text: '30 minutes',
+            value: '30 minutes',
+            localizationId: 'thirty_minutes',
+        },
+        'one-hour': {
+            text: '1 hour',
+            value: '1 hour',
+            localizationId: 'one_hour',
+        },
+        'four-hours': {
+            text: '4 hours',
+            value: '4 hours',
+            localizationId: 'four_hours',
+        },
+        today: {
+            text: 'Today',
+            value: 'Today',
+            localizationId: 'today',
+        },
+        'this-week': {
+            text: 'This week',
+            value: 'This week',
+            localizationId: 'this_week',
+        },
+        'date-and-time': {
+            text: 'Choose date and time',
+            value: 'Date and Time',
+            localizationId: 'choose_date_and_time',
+        },
+    };
+
+    return (
+        <div className='statusExpiry'>
+            <div className='statusExpiry__content'>
+                <MenuWrapper
+                    className={'statusExpiry__menu'}
+                >
+                    <span className='expiry-wrapper expiry-selector'>
+                        <FormattedMessage
+                            id='expiry_dropdown.clear_after'
+                            defaultMessage='Clear after: '
+                        />
+                        <span className='expiry-value'>
+                            {expiryMenuItems[expiry].value}
+                        </span>
+                        <span>
+                            <i
+                                className='fa fa-angle-down'
+                                aria-hidden='true'
+                            />
+                        </span>
+                    </span>
+                    <Menu
+                        ariaLabel={localizeMessage('expiry_dropdown.menuAriaLabel', 'Clear after')}
+                        id='statusExpiryMenu'
+                    >
+                        <Menu.Group>
+                            {Object.keys(expiryMenuItems).map((item, index) => (
+                                <Menu.ItemAction
+                                    key={index}
+                                    onClick={(event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => handleExpiryChange(event, item)}
+                                    ariaLabel={localizeMessage(`expiry_dropdown.${expiryMenuItems[item].localizationId}`, expiryMenuItems[item].text).toLowerCase()}
+                                    text={localizeMessage(`expiry_dropdown.${expiryMenuItems[item].localizationId}`, expiryMenuItems[item].text)}
```

Comment:
1. Using template string for localization ids causes the i18n script to remove these strings. Instead, we should define the entire ID in the `expiryMenuItems` constant.

2. Using the same ariaLabel and text seems redundant. Check if that is recommended practice for a11y.